.evergreen/run-async-std-tests.sh

```diff
@@ -4,3 +4,4 @@ set -o errexit
 
 . ~/.cargo/env
 RUST_BACKTRACE=1 cargo test --no-default-features --features async-std-runtime
+RUST_BACKTRACE=1 cargo test sync --no-default-features --features sync
```

Comment:
the tests are so minimal right now that I figured it wasn't worth a whole evergreen matrix, especially if we're only doing async-std for now.